v2/event/extensions.go

```diff
@@ -21,4 +20,14 @@ func caseInsensitiveSearch(key string, space map[string]interface{}) (interface{
 	return nil, false
 }
 
-var IsAlphaNumeric = regexp.MustCompile(`^[a-zA-Z0-9]+$`).MatchString
+func IsExtensionNameValid(key string) bool {
+	if len(key) < 1 || len(key) > 20 {
+		return false
+	}
+	for _, c := range key {
+		if !((c >= 'a' && c <= 'z') || (c >= '0' && c <= '9')) {
```

Comment:
A-Z is allowed and ignored, this change will be breaking for anyone who is using mixed-case keys and letting the sdk lowercase it.

Reply:
Should we keep the uppercase behaviour or remove it? If we want to keep it, then i'll just add it to the condition here